samza-core/src/main/scala/org/apache/samza/system/SystemConsumers.scala

```diff
@@ -219,6 +231,7 @@ class SystemConsumers (
     // If elasticity is enabled then the RunLoop gives SSP with keybucket
     // but the MessageChooser does not know about the KeyBucket
     // hence, use an SSP without KeyBucket
+    sspKeyBucketsRegistered.add(ssp)
```

Comment:
should this be conditional on if the elascticity factor > 1 ?
otherwise, the removeKeyBucket stuff can be skipped?

Reply:
`sspKeyBucketsRegistered` is used in one place only - where the choseObject is incremented. is the comment to not have `sspKeyBucketsRegistered` if elasticity factor = 1? With elasticity factor > 1, this book keeping is needed for correct counting of chose-object

removeKeyBucket is needed when elasticity > 1. it is currently used in 4 places and without this it would mean an if then else check in all the places.

Reply:
udpated to use condition when correcting choseObejct metric